pkg/collector/corechecks/snmp/status/status.go

```diff
@@ -48,6 +53,72 @@ func (Provider) populateStatus(stats map[string]interface{}) {
 		json.Unmarshal(snmpProfileErrorsJSON, &profiles) //nolint:errcheck
 		stats["snmpProfiles"] = profiles
 	}
+
+	autodiscoveryVar := expvar.Get("snmpAutodiscovery")
+
+	if autodiscoveryVar != nil {
+
+		var autodiscoveryConfig snmp.ListenerConfig
+		if pkgconfigsetup.Datadog().IsSet("network_devices.autodiscovery") {
+			err := pkgconfigsetup.Datadog().UnmarshalKey("network_devices.autodiscovery", &autodiscoveryConfig)
+			if err != nil {
+				return
+			}
+			var subnetOrder []string
+			for _, autodiscoveryConfig := range autodiscoveryConfig.Configs {
+				if autodiscoveryConfig.NetworkLegacy != "" {
+					subnetOrder = append(subnetOrder, autodiscoveryConfig.NetworkLegacy)
+				} else {
+					subnetOrder = append(subnetOrder, autodiscoveryConfig.Network)
+				}
+			}
+
+			autodiscoverySubnets := getSubnetsStatus(autodiscoveryVar)
+			var orderedSubnets []subnetStatus
+			for _, subnet := range subnetOrder {
+				for _, status := range autodiscoverySubnets {
+					if status.Subnet == subnet {
+						orderedSubnets = append(orderedSubnets, status)
+					}
+				}
+			}
```

Comment:
in case you have multiple subnets, there will be several lines indeed, but I think this is the expected behavior as you can have different configs for each subnet and so a different autodiscovery status

but okay to add a test